app/assets/javascripts/discourse/app/routes/topic-by-slug-or-id.js

```diff
@@ -12,6 +12,6 @@ export default DiscourseRoute.extend({
   },
 
   afterModel(result) {
-    DiscourseURL.routeTo(result.url, { replaceURL: true });
```

Comment:
This seems backwards to me... `replaceURL: true` should **prevent** the URL being kept in history. The default (`replaceURL: false`) should add a history entry. Do we have a bug deeper in the DiscourseURL module?

Reply:
That is what I found weird about this too. I suspect it has something to do with the fact that the redirect happens while the URL is loading.